ts41/utilities.d.ts

```diff
@@ -0,0 +1,8 @@
+/**
+Recursively split a string literal into two parts on the first occurence of the given string, returning an array literal of all the separate parts.
+*/
+export type Split<S extends string, D extends string> =
```

Comment:
This isn't really a "utility" as they are used in source/utilities.d.ts. It's more of a separate type – and as such deserves a separate file, description, use case and needs to be included in readme.

Reply:
👍 I'll mimic what you've done in your PR then it won't matter which goes in first (I won't get a chance to push another revision until Monday, most likely)

Reply:
I don't agree with the premise that all individual non-utility types should be exposed. We should expose what's generally useful. In this specific case, I do agree `Split` could be useful for people.

Reply:
In the end I left it here unexposed, because I don't want to add scope/keep in sync with a different PR, even if the eventual plan is to expose and document it. @kainiedziela maybe whichever of us merges in second can just deal with conflicts, should be v simple.